common/segments/serializers.py

```diff
@@ -249,16 +256,31 @@ def _update_or_create_segment_rule(
 
     @staticmethod
     def _update_or_create_conditions(
-        conditions_data: dict, rule: models.Model, is_create: bool = False
+        conditions_data: dict,
+        rule: models.Model,
+        segment: models.Model | None = None,
```

Comment:
This is clearly out of the scope of this PR, but maybe we should define common abstract models in the `flagsmith-common` package.

Reply:
I'd be in favour of this.